src/infiniband.c

```diff
@@ -0,0 +1,347 @@
+/**
+ * collectd - src/infiniband.c
+ * Copyright 2020 NVIDIA Corporation
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ * Authors:
+ *   Luke Yeager <lyeager at nvidia.com>
+ **/
+
+#include "collectd.h"
+
+#include "plugin.h"
+#include "utils/common/common.h"
+#include "utils/ignorelist/ignorelist.h"
+
+#if !KERNEL_LINUX
+#error "No applicable input method."
+#endif
+
+#include <ctype.h>
+#include <glob.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+/* Configuration settings ****************************************************/
+
+static const char *config_keys[] = {
+    "Port",
+    "IgnoreSelected",
+};
+static int config_keys_num = STATIC_ARRAY_SIZE(config_keys);
+static ignorelist_t *ignorelist;
+
+/* Listing ports *************************************************************/
+
+static int ib_glob_ports(glob_t *g) {
+  return glob("/sys/class/infiniband/*/ports/*/state", GLOB_NOSORT, NULL, g);
+}
+
+static int ib_parse_glob_port(char *path, char **device, char **port) {
+  char *tok, *saveptr = NULL;
+  int j = 0;
+  *device = NULL;
+  *port = NULL;
+  tok = strtok_r(path, "/", &saveptr);
+  while (tok != NULL) {
+    if (j == 3)
```

Comment:
Minor feedback, could a variable be used here instead of 3?

Reply:
Sounds reasonable to me. Done.